app/src/test/java/com/codesoom/assignment/application/TaskServiceTest.java

```diff
@@ -0,0 +1,204 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.*;
+
+@DisplayName("TaskService 클래스")
+class TaskServiceTest {
+    static final String TASK_TITLE = "Test Title";
+    static final Long TASK_ID = 1L;
+    static final Long INVALID_ID = 12384L;
+    static final String UPDATED_TITLE= "Test Title Updated";
+    private TaskService service;
+
+    @BeforeEach
+    void setUp() {
+        service = new TaskService();
+
+    }
+
+    @Nested
+    @DisplayName("getTasks 메소드는")
+    class Describe_getTasks {
+        @Nested
+        @DisplayName("만약 task가 없다면")
+        class No_task {
+            @Test
+            @DisplayName("빈 리스트를 리턴한다.")
+            void it_return_empty_List() {
+                assertThat(service.getTasks()).isEqualTo(new ArrayList());
+            }
+
+        }
+
+        @Nested
+        @DisplayName("만약 task가 3개 있으면")
+        class Has_task {
+            final static int TASK_SIZE = 3;
+
+            @BeforeEach
+            void setup() {
+                Task task = new Task();
+                for (int i = 0; i < TASK_SIZE; i++) {
+                    task.setTitle("test title-" + i);
+                    service.createTask(task);
+                }
+            }
+
+            @Test
+            @DisplayName("크기 3의 리스트를 반환한다.")
+            void it_retruns_tasks() {
+                assertThat(service.getTasks()).hasSize(TASK_SIZE);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("getTask 메소드는")
+    class Describe_getTask {
+        @Nested
+        @DisplayName("만약 요청한 Id에 해당하는 task를 찾지 못하면")
+        class No_task {
+
+            @Test
+            @DisplayName("TasksNotFoundException을 발생시킨다.")
```

Comment:
예외가 발생하는 건 메소드를 관찰하는 사람 입장이죠. 이 테스트는 메소드가 주인공입니다.

```suggestion
            @DisplayName("TasksNotFoundException을 던진다.")
```

Reply:
그렇군요. 깊이 생각해보지 못하고 작성한 부분인데 통일성과 직관성을 위해 메소드를 중심으로 작성하는 것이 좋아보이네요. 수정하겠습니다!!

Reply:
한편으로는 우리가 예외를 던지는 코드를 작성할 때 쓰는 `throws`를 곱씹어보셨으면 좋겠어요. 정말로 던지는 거죠.